Kernel/Syscalls/execve.cpp

```diff
@@ -851,15 +887,20 @@ ErrorOr<void> Process::exec(NonnullOwnPtr<KString> path, Vector<NonnullOwnPtr<KS
 
     VERIFY(description->inode());
 
-    // Read the first page of the program into memory so we can validate the binfmt of it
-    char first_page[PAGE_SIZE];
-    auto first_page_buffer = UserOrKernelBuffer::for_kernel_buffer((u8*)&first_page);
-    auto nread = TRY(description->read(first_page_buffer, sizeof(first_page)));
+    // Read just the size of ELF header of the program into memory so we can start parsing it
+    // If it is #! interpreted file, the size of a ELF header should still suffice.
```

Comment:
Is this comment accurate? We will use `PAGE_SIZE` later anyways.

Reply:
It should suffice for finding the shebang sign. And `PAGE_SIZE` is just an hardcoded value... in theory we should be able to read more because assuming max path length is kinda bad.

Should I rewrite this to reflect that it is sufficient for finding the shebang sign?

Reply:
Done.

Reply:
The thing I was confused about is that the comment says "the size of a ELF header should suffice" instead of "PAGE_SIZE should suffice".